botocore/signers.py

```diff
@@ -86,7 +87,7 @@ def sign(self, operation_name, request):
             signature_version=signature_version, request_signer=self)
 
         # Sign the request if the signature version isn't None or blank
-        if signature_version:
+        if signature_version != botocore.UNSIGNED:
```

Comment:
Given it's now a sentinel, we should be using identity checks instead of equality check (`signature_version is not botocore.UNSIGNED`)